app/lib/core/services/database_provider/db_services.dart

```diff
@@ -0,0 +1,212 @@
+import 'dart:async';
+import 'package:connectivity/connectivity.dart';
+import 'package:mysql1/mysql1.dart';
+import 'package:app/core/models/user.dart';
+import 'package:app/core/models/animal_transport_record.dart';
+
+class DatabaseServices {
+  var networkStatus;
+  MySqlConnection dbConnect;
+
+  Future<bool> checkConnectivity() async {
```

Comment:
I really hope there's either a plugin for this, or Firestore provides this.

Reply:
I used connectivity plugin for this so it should work with firestore as well